packages/shared/src/server/repositories/observations.ts

```diff
@@ -976,8 +976,8 @@ export const getObservationMetricsForPrompts = async (
           prompt_version,
           min(start_time) AS first_observation,
           max(start_time) AS last_observation,
-          medianExact(usage_details['input']) AS median_input_usage,
-          medianExact(usage_details['output']) AS median_output_usage,
+          medianExact(arraySum(mapValues(mapFilter(x -> positionCaseInsensitive(x.1, 'input') > 0, usage_details)))) AS median_input_usage,
+          medianExact(arraySum(mapValues(mapFilter(x -> positionCaseInsensitive(x.1, 'output') > 0, usage_details)))) AS median_output_usage,
```

Comment:
can you try this query for large users on US?

Reply:
Tried this query for one of our largest US customers, query took less than 200ms:

```sql
SELECT
    type,
	SUM(arraySum (mapValues (mapFilter (x -> positionCaseInsensitive (x .1, 'input') > 0, usage_details))) AS input_usage),
	SUM(arraySum (mapValues (mapFilter (x -> positionCaseInsensitive (x .1, 'output') > 0, usage_details))) AS output_usage)
FROM
	observations FINAL
WHERE
	(type = 'GENERATION')
	AND start_time >= toDateTime ('2024-12-16')
	AND project_id = '<projectId>'
GROUP BY
	type
```